client/txn.go

```diff
@@ -356,99 +349,82 @@ func endTxnCall(commit bool, hasTrigger bool) proto.Call {
 	}
 }
 
-func (txn *Txn) exec(retryable func(txn *Txn) error) (err error) {
+func (txn *Txn) exec(retryable func(txn *Txn) error) error {
 	// Run retryable in a retry loop until we encounter a success or
 	// error condition this loop isn't capable of handling.
-	txn.execing = true
+	var err error
 	for r := retry.Start(txn.db.txnRetryOptions); r.Next(); {
-		txn.haveTxnWrite, txn.haveEndTxn = false, false // always reset before [re]starting txn
 		if err = retryable(txn); err == nil {
-			if !txn.haveEndTxn {
-				// If there were no errors running retryable, commit the txn. This
-				// may block waiting for outstanding writes to complete in case
-				// retryable didn't -- we need the most recent of all response
-				// timestamps in order to commit.
-				err = txn.Commit()
-			}
+			// If there were no errors running retryable, commit the txn. This
+			// may block waiting for outstanding writes to complete in case
```

Comment:
what I think it means is that when `retryable` fires off some requests in goroutines, then hopefully the command queue will have the `EndTransaction` call wait for it. But that's only when they're on the same `Range` and `retryable` really really shouldn't do something like that anyways, so unless I'm missing the point of this comment we should probably remove it.